Extensions/NumberFormatter.swift

```diff
@@ -15,32 +15,29 @@ extension NumberFormatter {
         return string(from: NSNumber(value: number))
     }
 
-    func string(from number: Double, unit: String, style: Formatter.UnitStyle = .medium) -> String? {
+    func string(from number: Double, unit: String, style: Formatter.UnitStyle = .medium, avoidLineBreaking: Bool = true) -> String? {
         guard let stringValue = string(from: number) else {
             return nil
         }
-
-        let format: String
+        
+        let separator: String
         switch style {
-        case .long, .medium:
-            format = LocalizedString(
-                "quantity-and-unit-space",
-                value: "%1$@ %2$@",
-                comment: "Format string for combining localized numeric value and unit with a space. (1: numeric value)(2: unit)"
-            )
+        case .long:
+            separator = " "
+        case .medium:
+            separator = avoidLineBreaking ? "\u{00a0}" : " "
         case .short:
             fallthrough
         @unknown default:
-            format = LocalizedString(
-                "quantity-and-unit-tight",
-                value: "%1$@%2$@",
-                comment: "Format string for combining localized numeric value and unit without spacing. (1: numeric value)(2: unit)"
-            )
+            separator = avoidLineBreaking ? "\u{2060}" : ""
         }
-
+        
+        let unit = avoidLineBreaking ? unit.replacingOccurrences(of: "/", with: "\u{2060}/\u{2060}") : unit
+        
         return String(
-            format: format,
+            format: "%1$@%2$@%3$@",
```

Comment:
It seems the localization was dropped?

Reply:
Yes, I did that intentionally, because all this string is is a format of <value><space><units>.  I don't know if this would vary in other languages, would it?

Reply:
That is kind of the point. If we don't know that all languages do it the same way, we should allow that they might do it differently. Given it was there originally, it should be kept. 

Is there a problem if it is localized?

Reply:
Nope.  Will do, thanks!